incubator/hnc/internal/validators/namespace.go

```diff
@@ -79,7 +79,26 @@ func (v *Namespace) checkForest(req *nsRequest) admission.Response {
 
 	ns := v.Forest.Get(req.ns.Name)
 
-	// Early exit to allow non-delete requests or if the namespace allows cascading deletion.
+	// Allow all create namespace requests except if the namespace name already exists in
+	// the external hierarchy (the default webhook already denies existing namespace names).
+	if req.op == v1beta1.Create {
+		for _, nm := range v.Forest.GetNamespaceNames() {
+			if _, ok := v.Forest.Get(nm).ExternalTreeLabels[req.ns.Name]; ok {
+				return deny(metav1.StatusReasonAlreadyExists, "The name is already used. Please us a different name.")
```

Comment:
Instead of "The name is already used," I think we need a more specific error message. For example, "The namespace name %q is reserved by the external hierarchy manager %q"?